### PR TITLE
implement Dbus isRegistered() methods

### PR DESCRIPTION
**Pull request by johnfreed**
isRegistered(number\<s\>) returns a boolean

isRegistered(numbers\<as\>) returns an array of Booleans

### CODE REVIEW

**Comment by johnfreed** [comment]
not exactly sure why strikethrough is appearing when I use the standard java-dbus type documentation (i.e., \<s\> for string and \<as\> for array of strings)

File: src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java [hunk 1]
```diff
@@ -388,8 +389,28 @@ public String getGroupName(final byte[] groupId) {
     }
 
     @Override
-    public boolean isRegistered() {
-        return true;
+    public boolean isRegistered(String number) {
+        List<Boolean> result = isRegistered(List.of(number));
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java:393]: Please use type inference, where possible
```suggestion
        var result = isRegistered(List.of(number));
```

File: man/signal-cli-dbus.5.adoc [hunk 2]
```diff
@@ -29,126 +29,252 @@ method(arg1<type>, arg2<type>, ...) -> return<type>
...
+Exceptions: None
 
-sendMessage(message<s>, attachments<as>, recipient<s>) -> timestamp<x>::
-sendMessage(message<s>, attachments<as>, recipients<as>) -> timestamp<x>::
-* message     : Text to send (can be UTF8)
-* attachments : String array of filenames to send as attachments (passed as filename, so need to be readable by the user signal-cli is running under)
-* recipient   : Phone number of a single recipient
-* recipients  : Array of phone numbers 
-* timestamp   : Can be used to identify the corresponding signal reply
+getGroupIdsBase64() -> groupList<as>::
```
Comment (AsamK) [inline man/signal-cli-dbus.5.adoc:115]: please remove unrelated documentation changes

File: src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java [hunk 3]
```diff
@@ -388,8 +389,28 @@ public String getGroupName(final byte[] groupId) {
...
+    @Override
+    public List<Boolean> isRegistered(List<String> numbers) {
+        List<Boolean> results = new ArrayList<Boolean> ();
+        if (numbers.isEmpty()) {
+            return results;
+        }
+        try {
+            Map<String, Pair<String, UUID>> registered;
+            registered = m.areUsersRegistered(new HashSet<String>(numbers));
+            for (String number : numbers) {
```
Comment (AsamK) [inline src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java:406]: you can move the for loop out of the try/catch block.
And maybe convert it to a stream/map/collect construct.

File: src/main/java/org/asamk/Signal.java [hunk 4]
```diff
@@ -81,7 +81,9 @@ long sendGroupMessageReaction(
             byte[] groupId, String name, List<String> members, String avatar
     ) throws Error.AttachmentInvalid, Error.Failure, Error.InvalidNumber, Error.GroupNotFound, Error.InvalidGroupId;
 
-    boolean isRegistered();
```
Comment (AsamK) [inline src/main/java/org/asamk/Signal.java:84]: please keep the `isRegistered()` method for backwards compatibility.

**Review by AsamK — CHANGES_REQUESTED** [review]

File: src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java (hunk 3 above)
Comment (johnfreed) [reply src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java]: The stream/map/collect construct is no doubt more efficient, but I find it intimidating, like some of the Perl code I encounter from time to time. Please feel free to change what I've submitted to use it.

File: src/main/java/org/asamk/Signal.java (hunk 4 above)
Comment (johnfreed) [reply src/main/java/org/asamk/Signal.java:84]: done
